File: Competitive Programming/UVA 573 The Snail/Main.py
while True:
    test_list = list(map(int, raw_input().split(" ")))
    h = test_list[0]
    if h == 0:
        break
    u = test_list[1]
    d = test_list[2]
    f = test_list[3]
    days = 0
    initialHeight = 0.0
    distanceClimbed = u
    heightAfterClimbing = u
    heightAfterSliding = heightAfterClimbing - d
    fatigueFactor = u * (f / 100.0)
    print days, " :: " , initialHeight, " :: " , distanceClimbed , " :: " , heightAfterClimbing , " :: " , heightAfterSliding
    while True:
        days = days + 1
        if (days == 1):
            pass
        else:
            initialHeight = heightAfterSliding
            distanceClimbed = distanceClimbed - fatigueFactor
            if distanceClimbed <= 0:
                distanceClimbed = 0
            
            heightAfterClimbing = initialHeight + distanceClimbed
            heightAfterSliding = heightAfterClimbing - d
            print days, " :: " , initialHeight, " :: " , distanceClimbed , " :: " , heightAfterClimbing , " :: " , heightAfterSliding
        if heightAfterClimbing > h:
            print "success on day" , days
            break
        
        if heightAfterSliding < 0:
            print "failure on day" , days
            break

File: Competitive Programming/UVA 573 The Snail/Main.java
import java.util.Scanner;
import java.text.*;
public class Main {
    public static void main(String args[]) {
        Scanner scan = new Scanner(System.in);
        while(scan.hasNext()) {
            int h = scan.nextInt();
            if (h == 0) {
                break;
            }
            float u = (float)scan.nextInt();
            float d = (float)scan.nextInt();
            float f = (float)scan.nextInt();
            int days = 0;
            float initialHeight = 0.0f;
            float distanceClimbed = u;
            float fatigueFactor = u * (f / 100.0f);
            float heightAfterClimbing = initialHeight + u;
            float heightAfterSliding = heightAfterClimbing - d;
            DecimalFormat df = new DecimalFormat("#.##");
            do {
                days++;
                if (days == 1) {

                } else {
                    initialHeight = heightAfterSliding;
                    distanceClimbed = Float.parseFloat(df.format(distanceClimbed - fatigueFactor));
                    if (distanceClimbed <= 0.0f) {
                        distanceClimbed = 0;
                    }
                    heightAfterClimbing = Float.parseFloat(df.format(initialHeight + distanceClimbed));
                    heightAfterSliding = Float.parseFloat(df.format(heightAfterClimbing - d));
                }
                System.out.println(days + " :: " + initialHeight + " :: " + distanceClimbed + " :: " + heightAfterClimbing + " :: " + heightAfterSliding);
                
                if (heightAfterClimbing > h) {
                    System.out.println("success on day " + days);
                    break;
                }
                // heightAfterSliding = Float.parseFloat(df.format(heightAfterClimbing - d));
                if (heightAfterSliding < 0) {
                    System.out.println("failure on day " + days);
                    break;
                }
            } while(true);
        }
        scan.close();
    }
}